albumentations/augmentations/transforms.py

```diff
@@ -2664,3 +2667,151 @@ def get_params_dependent_on_targets(self, params: Dict[str, Any]) -> Dict[str, A
 
     def get_transform_init_args_names(self) -> Tuple[str, str, str, str, str, str, str]:
         return "mean", "std", "gauss_sigma", "intensity", "cutout_threshold", "mode", "color"
+
+
+class UnpropShuffle(DualTransform):
+    """
+    Unproportional shuffle grid's cells on image.
+    Args:
+        grid ((int, int)): size of grid for splitting image.
+    Targets:
+        image, mask
+    Image types:
+        uint8, float32
+    """
+
+    def __init__(self, grid=(3, 3), always_apply=False, p=0.5):
+        super(UnpropShuffle, self).__init__(always_apply, p)
+        self.grid = grid
+        self.shuffled_ids = []
+
+    def apply(self, img, tiles=None, **params):
+        if tiles is None:
+            tiles = []
+        self.ids = self.shuffled_ids.copy()
+        return F.unprop_swap_tiles_on_image(img, tiles, self.shuffled_ids)
+
+    def apply_to_mask(self, img, tiles=None, **params):
+        if tiles is None:
+            tiles = []
+        return F.unprop_swap_tiles_on_image(img, tiles, self.shuffled_ids)
+
+    def apply_to_bboxes(self, bboxes, **params):
+        tiles = params["tiles"]
+        rows = params["rows"]
+        cols = params["cols"]
+        return F.unprop_bbox_transpose(bboxes, tiles, self.ids, rows, cols)
+
+    def get_transform_init_args_names(self):
+        return ()
+
+    def apply_to_keypoint(self, keypoint, **params):
+        return ()
+
+    def get_params_dependent_on_targets(self, params):
+        num_segments = params["num_segments"] if "num_segments" in params else 7
+        ratio = params["ratio"] if "ratio" in params else 4.0 / 3.0
+        # refinement_steps = params["refinement_steps"] if "refinement_steps" in params else 10
+
+        tolerance = ratio * 0.20  # tolerance for the ratio above 20
+
+        @dataclass
+        class Segm:
+            x: float
+            y: float
+            width: float
+            height: float
+
+            def __str__(self):
+                return f"[{self.x}, {self.y}, {self.width}, {self.height}]"
+
+        segms = []
+        img_height, img_width = params["image"].shape[:2]
+        segms.append(Segm(0, 0, img_width, img_height))
+
+        min_side = min(img_width, img_height) // 6  # minimal side of the rectangle
+
+        def refinement():
+            """
+            Refinement the rectangles to look more like the demanded ratio considering tolerance and prevent stretches.
+            """
+            at_least_one_invalid = False
+            index = 0
+
+            # We must work with the copy because "Split" alters the original list
+            segms_copy = segms.copy()
+
+            for segm in segms_copy:
```

Comment:
Use `enum` for index

Reply:
@Dipet I would like to ask you what do you mean by `enum`. You mean enumerate function?

Reply:
Replace
```python
index = 0

# We must work with the copy because "Split" alters the original list
segms_copy = segms.copy()

for segm in segms_copy:
```

with:
```python
# We must work with the copy because "Split" alters the original list
segms_copy = segms.copy()

for index, segm in enumerate(segms_copy):
```